Src/library/net/network/socket/network_socket.e

```diff
@@ -163,25 +163,57 @@ feature -- Status report
 		local
 			retval: INTEGER
 		do
-			retval := c_check_exception_with_timeout (descriptor, timeout)
+			retval := c_check_exception_with_timeout (descriptor, timeout_ns)
 			Result := (retval > 0)
 		end
 
 	timeout: INTEGER
 			-- Duration of timeout in seconds
+		obsolete
+			"Use `timeout_ns` using nanoseconds [2018-01-15]"
+		do
+			Result := (timeout_ns // one_second_in_nanoseconds).to_integer_32
+			if Result = 0 then
```

Comment:
A negative value is possible due to an overflow, so it might make sense to check that the result is non-negative.

Reply:
fixed